diff --git a/file_entity.pages.inc b/file_entity.pages.inc
index 9d6e359..ea5cafd 100644
--- a/file_entity.pages.inc
+++ b/file_entity.pages.inc
@@ -244,6 +244,7 @@ function file_entity_add_upload_step_fields($form, &$form_state, array $options
 function file_entity_usage_page($file) {
   $rows = array();
   $occured_entities = array();
+  $languages = language_list();
 
   // Determine all of the locations where a file is used, then loop through the
   // occurrences and filter out any duplicates.
@@ -268,14 +269,14 @@ function file_entity_usage_page($file) {
         $entity_uri = empty($entities[$entity_id]) ? NULL : entity_uri($entity_type, $entities[$entity_id]);
 
         // Link the label to the URI when possible.
-        if (!empty($entity_uri['path'])) {
-          $entity_label = l($label, $entity_uri['path']);
+        if (empty($entity_uri)) {
+          $entity = check_plain($label);
         }
         else {
-          $entity_label = check_plain($label);
+          $entity = l($label, $entity_uri['path'], array('language' => $languages[$entities[$entity_id]->language]));
         }
 
-        $rows[] = array($entity_label, $entity_type, $count);
+        $rows[] = array($entity, $entity_type, $count);
 
         // Record the occurrence of the entity to ensure that it isn't listed in
         // the table again.
