diff --git a/modules/media_bulk_upload/media_bulk_upload.info b/modules/media_bulk_upload/media_bulk_upload.info
index 47b6463..da70576 100644
--- a/modules/media_bulk_upload/media_bulk_upload.info
+++ b/modules/media_bulk_upload/media_bulk_upload.info
@@ -5,10 +5,8 @@ core = 7.x
 
 dependencies[] = media
 dependencies[] = multiform
-dependencies[] = plupload
 
 test_dependencies[] = multiform
-test_dependencies[] = plupload
 
 files[] = includes/MediaBrowserBulkUpload.inc
 files[] = tests/media_bulk_upload.test
